src/NUnitFramework/tests/Attributes/TimeoutTests.cs

```diff
@@ -244,6 +248,23 @@ public void TestThatTimesOutIsRanToCompletionWhenDebuggerIsAttachedBeforeTimeOut
             Assert.That(_testRanToCompletion, () => "Test did not run to completion");
         }
 
+#if !NET35
+        private static readonly CancellationToken CancelledToken = new CancellationToken(true);
+        private static readonly CancellationToken PendingToken = new CancellationToken(false);
+
+        static readonly object[] TokenTestCases = new object[]
+        {
+            new object[] { true, CancelledToken },
+            new object[] { false, PendingToken },
+        };
+
+        [TestCaseSource(nameof(TokenTestCases))]
```

Comment:
👍thanks for adding this, there can be some interesting edge cases when dealing with parameterized tests.
Out of curiousity, does `TestCaseAttribute` work with this pattern of injecting a `CancellationToken` too?

Separate question: `TestCaseAttribute` also supports targeting test methods with a `params[]` array as the last method argument. Since can't have _both_ a `CancellationToken` and a `params object[]` as the last args, how would cooperative cancellation look there? Would a test have to read the `CancellationToken` from the `TestContext` instead?

Something like this pseudocode?
```cs
[TestCase(1,2,3, ExpectedResult = 6)]
public async Task<int> ParameterizedSum(params int[] numbers) {
  return await TestApi.SumAsync(numbers, TestContext.Current.CancellationToken);`
}
```